benchopt/base.py

```diff
@@ -385,16 +384,32 @@ def compute(self, beta):
             returned, it should at least contain a key `value` associated to a
             scalar value which will be used to detect convergence. With a
             dictionary, multiple metric values can be stored at once instead
-            of runnning each separately.
+            of running each separately.
         """
         ...
 
-    def __call__(self, beta):
-        """Used to call the computation of the objective.
+    def __call__(self, solver_result):
+        """Used to call the evaluation of the objective.
 
         This allow to standardize the output to a dictionary.
         """
-        objective_dict = self.compute(beta)
+        # XXX remove in version 1.5
+        if hasattr(self, "compute"):
+            warnings.warn(
+                "objective.compute was renamed `objective.evaluate_result` in "
+                "v 1.4", FutureWarning,
+            )
+            self.evaluate_result = self.compute
+        # XXX remove in version 1.5
+        try:
+            objective_dict = self.evaluate_result(**solver_result)
```

Comment:
We better write

```python
try:
    kwargs = {**solver_result}
except TypeError:
    warnings.warn("...")
```
to avoid catching potential `TypeErrors` that might occur in `self.evaluate_result`.

WDTY @mathurinm?